src/uct/ib/mlx5/ib_mlx5.inl

```diff
@@ -175,6 +175,44 @@ uct_ib_mlx5_inline_copy(void *restrict dest, const void *restrict src, unsigned
 }
 
 
+static UCS_F_ALWAYS_INLINE void
+uct_ib_mlx5_inline_iov_copy(void *restrict dest, const uct_iov_t *iov, size_t iovcnt,
+                            unsigned length, uct_ib_mlx5_txwq_t *wq)
+{
+    size_t iov_it;
+    ptrdiff_t n;
+
+    ucs_assert(dest != NULL);
+    ucs_assert((iov != NULL) || (iovcnt == 0));
+
+    if (UCS_PTR_BYTE_OFFSET(dest, length) <= wq->qend) {
+        for (iov_it = 0; iov_it < iovcnt; ++iov_it) {
+            memcpy(dest, iov[iov_it].buffer, iov[iov_it].length);
+            dest = UCS_PTR_BYTE_OFFSET(dest, iov[iov_it].length);
+        }
+    } else {
+        n = UCS_PTR_BYTE_DIFF(dest, wq->qend);
+
+        for (iov_it = 0; iov[iov_it].length < n; ++iov_it) {
```

Comment:
looks like something is wrong here for the copying data from IOV to the WQ's `qend`
should it be smth like below?
```
for (iov_it = 0; (iov_it < iovcnt) && (n != 0); ++iov_it) {
    copy_length = ucs_min(iov[iov_it].length, n);
    memcpy(dest, iov[iov_it].buffer, copy_length);
    dest        = UCS_PTR_BYTE_OFFSET(dest, copy_length);
    n          -= copy_length;
}

dest        = wq->qstart;
copy_length = iov[iov_it].length - n;
memcpy(dest, UCS_PTR_BYTE_OFFSET(iov[iov_it].buffer, n), copy_length);
dest        = UCS_PTR_BYTE_OFFSET(dest, copy_length);

for (++iov_it; iov_it < iovcnt; ++iov_it) {
    memcpy(dest, iov[iov_it].buffer, iov[iov_it].length);
    dest = UCS_PTR_BYTE_OFFSET(dest, iov[iov_it].length);
}
```

Reply:
There is a ucs_min inside the first loop in your implementation. The ucs_min adds a comparison to the loop body. Thus I check that iov[iov_it].length is less than space before the end.

Reply:
👍